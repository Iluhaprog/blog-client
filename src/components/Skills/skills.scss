@import '../../vars.scss';

.skills {
    &-title {
        font-size: 20px;
        margin-right: 15px;
        font-weight: 300;
        border-bottom: 1px solid #000;
    }

    li {
        margin-bottom: 10px;
    }
}