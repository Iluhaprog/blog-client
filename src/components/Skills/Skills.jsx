import React from 'react';
import PropTypes from 'prop-types';
import { TagBox } from '../boxes';
import './skills.scss';

function Skills(props) {
    return (
        <ul className='row row_w-w skills'>
            <li className='skills-title'>
                Skills: 
            </li>
            {
                props.skills.map(skill => (
                    <TagBox 
                        key={skill}
                        title={skill}
                        mode='violet'
                    />
                ))
            }
        </ul>
    );
}

Skills.defaultProps = {
    skills: [],
};

Skills.propTypes = {
    skills: PropTypes.array,
};

export default Skills;