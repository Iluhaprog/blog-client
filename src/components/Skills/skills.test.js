import React from 'react';
import { render } from '@testing-library/react';
import Skills from './Skills';

test('Should render Skills component', () => {
    const { asFragment } = render(<Skills skills={['test']} />);
    expect(asFragment()).toMatchSnapshot();
});