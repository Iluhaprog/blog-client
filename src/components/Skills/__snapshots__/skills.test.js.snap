// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Skills component 1`] = `
<DocumentFragment>
  <ul
    class="row row_w-w skills"
  >
    <li
      class="skills-title"
    >
      Skills:
    </li>
    <li
      class="tag tag__violet"
    >
      test
    </li>
  </ul>
</DocumentFragment>
`;
